docs/concepts/crds/backupsession.md

```diff
@@ -0,0 +1,139 @@
+---
+title: BackupSession Overview
+menu:
+  product_stash_0.8.3:
+    identifier: backupsession-overview
+    name: BackupSession
+    parent: crds
+    weight: 20
+product_name: stash
+menu_name: product_stash_0.8.3
+section_menu_id: concepts
+---
+
+> New to Stash? Please start [here](/docs/concepts/README.md).
+
+# BackupSession
+
+## What is BackupSession
+
+A `BackupSession` is a Kubernetes `CustomResourceDefinition`(CRD) which represents a backup instance of the respective target pointed by a `BackupConfiguration` in a Kubernetes native way.
+
+When a backup schedule appears, the respective `CronJob` creates a `BackupSession` object. It points to the respective `BackupConfiguration`. Controller that runs inside backup sidecar (in case of backup through job, it is stash operator itself) watches this `BackupSession` object and start taking backup instantly.
+
+You can also create a `BackupSession` object manually to trigger backup instantly.
+
+>In this tutorial, we are going to use **host** word to indicate an entity (pod) whose data is backed up.
+
+>- For `Deployment`, `ReplicationController` and `ReplicaSet`, backup process run in only one pod. This pod is referred as **host-0**.
+>- For `StatefulSet`, backup process run in all pods. In this case, **pod-0** is known as **host-0**, **pod-1** is known as **host-1**, **pod-2** is known as **host-2** and so on.
+>- For `DaemonSet`, backup process run in all daemon pods. In this case, the **node name** where the pod is running act as their **host** name.
+>- For database or stand-alone PVC, backup is done by a job. In this case, the backup job's pod is known as **host-0**.
+
+## BackupSession CRD Specification
+
+Like any official Kubernetes resource, a `BackupSession` has `TypeMeta`, `ObjectMeta` and `Spec` , `Status` sections.
+
+A sample `BackupSession` created for backing up a Deployment's data is shown below,
+
+```yaml
+apiVersion: stash.appscode.com/v1beta1
+kind: BackupSession
+metadata:
+  name: deployment-backup-2do7af
+  namespace: demo
+  labels:
+    backup-configuration: deployment-backup
+spec:
+  backupConfiguration:
+    name: deployment-backup
+status:
+  phase: Succeeded
+  totalHosts: 1
+  sessionDuration: 3.861369766s
+  stats:
+  - hostname: host-0
+    phase: Succeeded
+    duration: 3.861369766s
+    snapshots:
+    - directory: /source/data
+      fileStats:
+        modifiedFiles: 0
+        newFiles: 0
+        totalFiles: 2
+        unmodifiedFiles: 2
+      name: ea5381cc
+      processingTime: 0m0s
+      size: 30 B
+      uploaded: 0 B
+```
+
+Here, we are going to describe some important sections of `BackupSession` crd.
+
+### BackupSession `Metadata`
+
+#### metadata.name
+
+`metadata.name` indicates the name of the `BackupSession`. This name is automatically generated by respective `CronJob` and it follows the following pattern: `<BackupConfiguration name>-<random suffix>`.
```

Comment:
Can we use the format that CronJobs use to generate names for Jobs for a CronJob; eg, backupconfig-{TIMESTAMP _ upto sec?}